### PR TITLE
MBS-10666 new test runner gradle plugin; gradle plugin publication poc

### PR DESCRIPTION
**Pull request by dsvoronin**
Plan is to apply one or another plugin based on feature flag
Both plugin will reuse same logic extracted to `runner-client` module

### CODE REVIEW

File: subprojects/gradle/cd/build.gradle.kts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 plugins {
     id("convention.kotlin-jvm")
-    id("convention.publish-gradle-plugin")
+    id("convention.publish-gradle-plugin-legacy")
```
Comment (sergioboi) [inline subprojects/gradle/cd/build.gradle.kts:3]: I don't get what is the purpose of that plugin?

Comment (dsvoronin) [reply subprojects/gradle/cd/build.gradle.kts:3]: convention for publication of gradle plugins;

problem is here: https://github.com/avito-tech/avito-android/pull/791/files#diff-e7b95c7356270a4bf48fea167dfbb47a8fa67e7a8ab35930fce58a7e9c047f2cR12

trying to publish new plugin correctly with new convention along with legacy one

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/android/testrunner/service/TestService.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.avito.android.testrunner.service
+
+import com.avito.android.testrunner.service.TestService.Params
+import com.avito.utils.gradle.KubernetesCredentials
+import org.gradle.api.provider.Property
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters
+
+@Suppress("UnstableApiUsage")
+public abstract class TestService : BuildService<Params>, AutoCloseable {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/android/testrunner/service/TestService.kt:10]: It'd easier to see this change in another PR. It doesn't related to plugins publication

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/android/testrunner/service/TestService.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/android/testrunner/service/TestService.kt:10]: ok, i'll extract publication logic

**Comment by dsvoronin** [comment]
will do another one

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/android/testrunner/service/TestService.kt (hunk 2 above)
Comment (eugene-krivobokov) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/android/testrunner/service/TestService.kt:10]: Thanks

**Pull request closed without merging**